From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MichaelSimons <msimons@microsoft.com>
Date: Sat, 19 Feb 2022 12:12:32 +0000
Subject: [PATCH] Define SystemReflectionMetadataLoadContextVersion property
 for package references

System.Reflection.MetadataLoadContext references were using an incorrect version.props property which breaks source-build.
In 6.0.3 SystemReflectionMetadataVersion was revved but not SystemReflectionMetadataLoadContextVersion therefore breaking
the build because a System.Reflection.MetadataLoadContext with the SystemReflectionMetadataVersion does not exist.
---
 eng/Versions.props                             | 1 +
 src/tasks/WasmAppBuilder/WasmAppBuilder.csproj | 4 ++--
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/eng/Versions.props b/eng/Versions.props
index 5e9d0039096..34034aa87ba 100644
--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -86,6 +86,7 @@
     <SystemReflectionMetadataVersion>5.0.0</SystemReflectionMetadataVersion>
     <!-- Static version does not get updated with the latest version during source-builds. -->
     <SystemReflectionMetadataStaticVersion>5.0.0</SystemReflectionMetadataStaticVersion>
+    <SystemReflectionMetadataLoadContextVersion>5.0.0</SystemReflectionMetadataLoadContextVersion>
     <SystemReflectionEmitVersion>4.7.0</SystemReflectionEmitVersion>
     <SystemReflectionEmitILGenerationVersion>4.7.0</SystemReflectionEmitILGenerationVersion>
     <SystemReflectionEmitLightweightVersion>4.7.0</SystemReflectionEmitLightweightVersion>
diff --git a/src/tasks/WasmAppBuilder/WasmAppBuilder.csproj b/src/tasks/WasmAppBuilder/WasmAppBuilder.csproj
index 358de5541e3..abf6d26d9ef 100644
--- a/src/tasks/WasmAppBuilder/WasmAppBuilder.csproj
+++ b/src/tasks/WasmAppBuilder/WasmAppBuilder.csproj
@@ -23,11 +23,11 @@
   </ItemGroup>
 
   <ItemGroup Condition="'$(TargetFramework)' == '$(TargetFrameworkForNETCoreTasks)'">
-    <PackageReference Include="System.Reflection.MetadataLoadContext" Version="$(SystemReflectionMetadataVersion)" />
+    <PackageReference Include="System.Reflection.MetadataLoadContext" Version="$(SystemReflectionMetadataLoadContextVersion)" />
   </ItemGroup>
 
   <ItemGroup Condition="'$(TargetFramework)' == '$(TargetFrameworkForNETFrameworkTasks)'">
-    <PackageReference Include="System.Reflection.MetadataLoadContext" Version="$(SystemReflectionMetadataVersion)" PrivateAssets="all" />
+    <PackageReference Include="System.Reflection.MetadataLoadContext" Version="$(SystemReflectionMetadataLoadContextVersion)" PrivateAssets="all" />
 
     <!-- match was MonoTargets tasks have -->
     <!-- These versions should not be newer than what Visual Studio MSBuild uses -->
